pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2300,6 +2300,12 @@ protected void internalGetMessageById(AsyncResponse asyncResponse, long ledgerId
             }
             PersistentTopic topic = (PersistentTopic) getTopicReference(topicName);
             ManagedLedgerImpl ledger = (ManagedLedgerImpl) topic.getManagedLedger();
+            if (null == ledger.getLedgerInfo(ledgerId).get()) {
+                log.error("[{}] Failed to get message with ledgerId {} entryId {} from {}, "
+                                + "ledgerId not belong to this topic.",
+                        clientAppId(), ledgerId, entryId, topicName);
+                asyncResponse.resume(new RestException(Status.NOT_FOUND, "Message not found"));
```

Comment:
It seems that the response message is not clear, the message with a specific position does not exist or the ledger does not belong to this topic?

Reply:
OK. I will change it.